src/main/scala/tilelink/TLController.scala

```diff
@@ -60,11 +60,17 @@ trait HasTLController {
 
     private def inputCrossing(x: ControlInput): RegField = {
         val width = x.width
-        val (toClock, toReset) = x.clock match {
-            case OuterClock => (this.clock, this.reset.toBool)
-            case TxClock => (this.io.txClock, this.io.txReset)
-            case RxClock => (this.io.rxClock, this.io.rxReset)
-            case _ => ???
+        val (toClock, toReset) = this match {
+            case l: Lane => x.clock match {
+                case OuterClock => (l.clock, l.reset.toBool)
+                case TxClock => (l.io.txClock, l.io.txReset)
+                case RxClock => (l.io.rxClock, l.io.rxReset)
+                case _ => ???
+            }
+            case _ => x.clock match {
+                case OuterClock => (this.clock, this.reset.toBool)
+                case _ => ???
+            }
```

Comment:
IMO, if we're genericizing this, these probably need to change (Your random analog widget might have a different set of clocks)